app/src/org/commcare/preferences/HiddenPreferences.java

```diff
@@ -449,25 +448,25 @@ public static boolean shouldBypassPreUpdateSync() {
         return CommCareApplication.instance().getCurrentApp().getAppPreferences().getBoolean(BYPASS_PRE_UPDATE_SYNC, false);
     }
 
-    public static void skipCommCareVersionForUpdate(int version) {
-        CommCareApplication.instance().getCurrentApp().getAppPreferences()
+    public static void setLastInAppUpdateCheckTime(long millis) {
+        PreferenceManager.getDefaultSharedPreferences(CommCareApplication.instance())
                 .edit()
-                .putInt(SKIP_COMMCARE_VERSION_FOR_UPDATE, version)
+                .putLong(LAST_IN_APP_UPDATE_CHECK_TIME, millis)
                 .apply();
     }
 
-    public static int getSkippedCommCareUpdateVersion() {
-        return CommCareApplication.instance().getCurrentApp().getAppPreferences().getInt(SKIP_COMMCARE_VERSION_FOR_UPDATE, -1);
+    public static long getLastInAppUpdateCheckTime() {
+        return PreferenceManager.getDefaultSharedPreferences(CommCareApplication.instance())
+                .getLong(LAST_IN_APP_UPDATE_CHECK_TIME, -1);
     }
 
-    public static void setLastInAppUpdateCheckTime(long millis) {
-        CommCareApplication.instance().getCurrentApp().getAppPreferences()
-                .edit()
-                .putLong(LAST_IN_APP_UPDATE_CHECK_TIME, millis)
-                .apply();
+    public static void cancelCommCareUpdate(String version) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CommCareApplication.instance());
+        int count = sharedPreferences.getInt(version, 0);
```

Comment:
we should prefix the key with a constant to make it more explicit for eg - `COMMCARE_UPDATE_CANCELLATION_COUNTER + "_" + version`